src/igraph/operators.py

```diff
@@ -0,0 +1,482 @@
+# vim:ts=4:sw=4:sts=4:et
+# -*- coding: utf-8 -*-
+"""
+IGraph library.
+
+@undocumented: deprecated, _graphmethod, _add_proxy_methods, _layout_method_wrapper,
+               _3d_version_for
+"""
+
+from __future__ import with_statement
+
+__license__ = u"""
+Copyright (C) 2006-2012  Tamás Nepusz <ntamas@gmail.com>
+Pázmány Péter sétány 1/a, 1117 Budapest, Hungary
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 2 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program; if not, write to the Free Software
+Foundation, Inc.,  51 Franklin Street, Fifth Floor, Boston, MA
+02110-1301 USA
+"""
+
+# pylint: disable-msg=W0401
+# W0401: wildcard import
+from igraph._igraph import *
+from igraph._igraph import _union, _intersection, _disjoint_union
+
+from collections import defaultdict, Counter
+from warnings import warn
+
+
+def disjoint_union(graphs):
+    """Graph disjoint union.
+
+    The disjoint union of two or more graphs is created.
+
+    This function keeps the attributes of all graphs. All graph, vertex and
+    edge attributes are copied to the result. If an attribute is present in
+    multiple graphs and would result a name clash, then this attribute is
+    renamed by adding suffixes: _1, _2, etc.
+
+    An error is generated if some input graphs are directed and others are
+    undirected.
+
+    @param graph: list of graphs. A lazy sequence is not acceptable.
+    @return: the disjoint union graph
+    """
+    if any(not isinstance(g, GraphBase) for g in graphs):
+        raise TypeError('Not all elements are graphs')
+
+    ngr = len(graphs)
+    # Trivial cases
+    if ngr == 0:
+        return None
```

Comment:
I'd rather throw an exception here. My first instinct would have been to return an empty graph, but we cannot decide whether it should be directed or undirected. An exception makes it clear that this is an ambiguous situation; furthermore, the `min()` and `max()` functions in Python also throw an exception for empty sequences so we are consistent with the rest of Python if we do it this way.

I'll take care of this today so we can merge this PR before the day starts for you :)